src/gl/metal/vcTexture.mm

```diff
@@ -14,75 +14,99 @@
 uint32_t g_textureIndex = 0;
 uint32_t g_blitTargetID = 0;
 
+void vcTexture_GetFormatAndPixelSize(const vcTextureFormat format, int *pPixelSize = nullptr, MTLPixelFormat *pPixelFormat = nullptr, MTLStorageMode *pStorageMode = nullptr, MTLTextureUsage *pUsage = nullptr)
+{
+  // defaults
+  MTLPixelFormat pixelFormat = MTLPixelFormatRGBA8Unorm;
+  MTLStorageMode storageMode = MTLStorageModeManaged; 
+  MTLTextureUsage usage = MTLTextureUsageShaderRead;
+  int pixelSize = 0; // in bytes
+
+#if UDPLATFORM_IOS || UDPLATFORM_IOS_SIMULATOR
+  pTextureDesc.storageMode = MTLStorageModeShared; // default on iOS and tvOS
+#endif
+
+  switch (format)
+  {
+  case vcTextureFormat_RGBA8:
+    pixelFormat = MTLPixelFormatRGBA8Unorm;
+    usage = MTLTextureUsagePixelFormatView | MTLTextureUsageShaderRead;
+    pixelSize = 4;
+#if UDPLATFORM_IOS || UDPLATFORM_IOS_SIMULATOR
+    storageMode = MTLStorageModeShared;
+#elif UDPLATFORM_OSX
+    storageMode = MTLStorageModeManaged;
+#endif
+    break;
+  case vcTextureFormat_BGRA8:
+    pixelFormat = MTLPixelFormatBGRA8Unorm;
+    usage = MTLTextureUsageShaderRead;
+    pixelSize = 4;
+#if UDPLATFORM_IOS || UDPLATFORM_IOS_SIMULATOR
+    storageMode = MTLStorageModeShared;
+#elif UDPLATFORM_OSX
+    storageMode = MTLStorageModeManaged;
+#endif
+    break;
```

Comment:
Storage mode for these types cannot be reliably predicted from vcTextureFormat, contrary to something I said at one point the render targets are all private, and any cpu reads we do on them atm are actually blits, are we ever calling this function to check render targets?

Reply:
I'm not sure what you mean here. This PR has simply re-organised code to remove duplication. No functionality has been added or removed. 

Are you suggesting the original code has problems? Or have I made a mistake in consolidating this functionality?

Reply:
Oh nvm, I was considering the function entirely out of context